api/explorer/api/test/test_ingest.py

```diff
@@ -0,0 +1,199 @@
+from collections import defaultdict
+from typing import Any, Callable, Mapping, cast
+from gremlin_python.driver import serializer
+from gremlin_python.driver.driver_remote_connection import DriverRemoteConnection
+from gremlin_python.process.anonymous_traversal import traversal
+
+import pytest
+
+from explorer.api import graph
+from explorer.core import model
+
+
+#######################################################################################
+# Mock Data
+#######################################################################################
+
+dummy_pkg_d = model.Package(
+    name="D",
+    output_paths={model.OutputPathName.OUT: "/D"},
+    nixpkgs_metadata=model.NixpkgsMetadata(
+        pname="D", version="1.0", broken=False, license="MIT"
+    ),
+    build_inputs=[],
+)
+
+
+dummy_pkg_c = model.Package(
+    name="C",
+    output_paths={model.OutputPathName.OUT: "/C"},
+    nixpkgs_metadata=model.NixpkgsMetadata(
+        pname="C",
+        version="1.0",
+        broken=False,
+        license="MIT",
+    ),
+    build_inputs=[],
+)
+
+dummy_pkg_b = model.Package(
+    name="B",
+    output_paths={model.OutputPathName.OUT: "/B"},
+    nixpkgs_metadata=model.NixpkgsMetadata(
+        pname="B", version="1.0", broken=False, license="MIT"
+    ),
+    build_inputs=[
+        model.BuildInput(
+            build_input_type=model.BuildInputType.BUILD_INPUT,
+            package=dummy_pkg_c,
+        )
+    ],
+)
+
+dummy_pkg_a = model.Package(
+    name="A",
+    output_paths={model.OutputPathName.OUT: "/A"},
+    nixpkgs_metadata=model.NixpkgsMetadata(
+        pname="A", version="1.0", broken=False, license="MIT"
+    ),
+    build_inputs=[
+        model.BuildInput(
+            build_input_type=model.BuildInputType.BUILD_INPUT,
+            package=dummy_pkg_b,
+        ),
+        model.BuildInput(
+            build_input_type=model.BuildInputType.BUILD_INPUT,
+            package=dummy_pkg_d,
+        ),
+    ],
+)
+
+dummy_nix_graph = model.NixGraph(packages=[dummy_pkg_a])
+
+#######################################################################################
+# Fixture(s)
+#######################################################################################
+@pytest.fixture
+def graph_connection():
+    conn = DriverRemoteConnection(
+        "ws://localhost:8182/gremlin",
+        "g",
+        message_serializer=serializer.GraphSONMessageSerializer(),
+    )
+    g = traversal().with_remote(conn)
+    yield conn
+    # Dropping all data from graph on teardown
+    g.V().drop().iterate()
+    g.E().drop().iterate()
+    conn.close()
```

Comment:
Unfortunately not really. Currently it will fail the first execution, cleanup existing data, then work on the second execution.
 
With Gremlin you operate on a graph by name, which here we call `g`. One option we could consider would be to add an additional graph which targets a different Postgres database (e.g. `gTest`). This could then specifically to be used for testing with docker-compose, though we might want to implement that in a separate PR.

Reply:
How about this?

```suggestion
@pytest.fixture
def graph_connection():
    conn = DriverRemoteConnection(
        "ws://localhost:8182/gremlin",
        # name of the graph
        # FIXME use a graph dedicated to test only
        "g",
        message_serializer=serializer.GraphSONMessageSerializer(),
    )
    g = traversal().with_remote(conn)
    yield conn
    # Dropping all data from graph on teardown
    g.V().drop().iterate()
    g.E().drop().iterate()
    conn.close()
```